MapboxSearch/base/src/main/java/com/mapbox/search/base/location/LocationEngineAdapter.kt

```diff
@@ -54,11 +68,21 @@ class LocationEngineAdapter(
 
     private fun startLocationListener() {
         locationProvider?.addLocationObserver(locationObserver)
+
+        locationEngineObservationTimeout?.let { timeout ->
+            timeoutWatcherJob?.cancel()
+            timeoutWatcherJob = CoroutineScope(Job()).launch {
```

Comment:
Coroutines seem a bit overcomplicated here. Using Android SDK's [Handler.postDelayed()](https://developer.android.com/reference/android/os/Handler#postDelayed(java.lang.Runnable,%20long)) would suffice.

Reply:
Frankly, I don't see any advantage. Moreover, it's tested by the customer.